platform-hub-api/app/models/concerns/validate_hashes.rb

```diff
@@ -0,0 +1,111 @@
+module ValidateHashes
+  extend ActiveSupport::Concern
+
+  # Uses https://github.com/deseretbook/classy_hash for the schema definition
+  # and validation.
+  #
+  # NOTE: Doesn't currently do strict matching of schema to data. This means
+  # your Hash can contain fields not defined in the schema, and only those
+  # defined in the schema are actually checked. This is more lenient and
+  # forgiving and allows you to do partial validation of hashes, which is
+  # essential for forward compatability. Strict check _can_ be easily added
+  # using the `strict: true` option for ClassyHash, but we should probably not
+  # do this!
+
+  CONFIG_ENTRY_ALLOWED_KEYS = [
+    :schema,
+    :unique_checks
+  ]
+
+  class_methods do
+
+    # Example usage:
+    #
+    #   validate_hashes(
+    #     field_one: {
+    #       schema: <ClassyHash schema>,
+    #       unique_checks: [
+    #         { array_path: [ :foo, :bar ], obj_key: :id },
+    #         { array_path: :baz }
+    #       ]
+    #     },
+    #     field_two: { schema: <ClassyHash schema> }
+    #   )
+    #
+    # NOTE: any properties used in the `unique_checks` (via the `array_path`)
+    # are expected to be required properties, and so will cause an error if nil
+    # in the actual data (regardless of whether they are marked as :optional /
+    # NilClass in the ClassyHash schema).
+    def validate_hashes config
+      mattr_accessor :validate_hashes_config, instance_writer: false
+
+      validate :validate_hashes_validation
+
+      # Check that config provided is valid
+      config_ok = config.is_a?(Hash) &&
+        config.keys.any? &&
+        config.values.all? { |h| (h.keys - CONFIG_ENTRY_ALLOWED_KEYS).empty? }
+
+      unless config_ok
+        raise 'Invalid config provided to validate_hashes'
```

Comment:
Given this only runs at initial load time (i.e. when models are first loaded in memory), I figured it didn't need a specific exception, especially as it's a halting error (i.e. stops the server from loading at all).